src/exec.rs

```diff
@@ -6,7 +6,7 @@ use std::task::{Context, Poll, RawWaker, RawWakerVTable, Waker};
 use crate::Error;
 
 // Returns Pending as long as its inner bool is false.
-#[derive(Default)]
+#[derive(Default, Clone, Copy, Debug)]
```

Comment:
I did not verified so I don't really know, but is it really safe to do that for macroquad's futures? What implications does it bring?

Reply:
Yeah, I think I might've been too aggressive with the impls. Let me know if there's anything else unsound.